test/common/network/connection_impl_test.cc

```diff
@@ -797,6 +797,38 @@ TEST_P(ConnectionImplTest, BasicWrite) {
   disconnect(true);
 }
 
+// Validate dynamic socket options
+TEST_P(ConnectionImplTest, DynamicSocketOptions) {
+  setUpBasicConnection();
+  connect();
+
+  auto dynamic_sockopt_ptr = client_connection_->getDynamicSocketOptionsPtr();
+  {
+    // Test set and get socket receive buffer size
+    uint32_t set_recv_buf_size = 81920;
+    uint32_t get_recv_buf_size = 0;
+    dynamic_sockopt_ptr->setSocketRecvBufferSize(set_recv_buf_size);
```

Comment:
Please get first, to verify you're not setting it to the default value

Reply:
Got it. I will do a get and add 4096 bytes to get while I am setting. This way we will not need to choose a random values that can conflict with the get value.